code/go/0chain.net/smartcontract/storagesc/free_allocation.go

```diff
@@ -294,7 +294,7 @@ func (ssc *StorageSmartContract) freeAllocationRequest(
 			"subtracting read pool token from transaction value: %v", err)
 	}
 
-	resp, err := ssc.newAllocationRequestInternal(txn, arBytes, conf, WithTokenTransfer(writePoolTokens, assigner.ClientId, txn.ToClientID), balances, nil)
+	resp, err := ssc.newAllocationRequestInternal(txn, arBytes, conf, NewTokenTransfer(writePoolTokens, assigner.ClientId, txn.ToClientID, true), balances, nil)
```

Comment:
Seems we need to do this to read pool, the read pool lock below should do transfer too.